### PR TITLE
feat: implement expanded-mode route

### PR DESCRIPTION
**Pull request by samobasquiat**
# Summary

Implemented expanded-mode route.
It doesn't have a loading state & and a hover for each step yet.
And its dark theme might be changed. 

# How did you test this change?

It can be tested in the storybook.
For testing dark mode in the storybook, you can change the `defaultValue` of `globalTypes` 
Or you can easily pass `darkTheme` as a `storyTheme` in `withTheme`
both in in `widget/ui/.storybook/preview.tsx`

# Checklist:

- [x] I have performed a self-review of my code
- [x] Implemented a user interface (UI) change, referencing our Figma design to ensure pixel-perfect precision.

### CODE REVIEW

File: widget/ui/src/components/TagRoute/index.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export { TagRoute } from './TagRoute';
```
Comment (RanGojo) [inline widget/ui/src/components/TagRoute/index.ts:1]: You could change it to `QuoteTag`

File: widget/ui/src/components/ExpandedRoute/ExpandedRoute.types.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import type { SwapInputProps } from 'src/containers/SwapInput/SwapInput.types';
+
+export interface PropTypes {
+  steps: Step[];
+  tags: RouteTag[];
+  time: QuoteCostProps['time'];
+  fee: QuoteCostProps['fee'];
+  percentageChange?: PriceImpactProps['percentageChange'];
+  warningLevel?: PriceImpactProps['warningLevel'];
+  outputPrice: SwapInputProps['price'];
```
Comment (RanGojo) [inline widget/ui/src/components/ExpandedRoute/ExpandedRoute.types.ts:14]: I'm not sure. Isn't better to move this component to widget embedded component and just pass `quote: SelectedQuote` as a props?

```
export interface PropTypes {
  quote: SelectedQuote
  tooltipContainer?: HTMLElement;
  onClick?: () => void;
  selected?: boolean;
}
```

File: widget/ui/src/components/ExpandedRoute/ExpandedRoute.tsx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+  RouteHeader,
+  Step,
+  Steps,
+  StyledPriceImpact,
+  SwapperContainer,
+  TagsContainer,
+  VerticalLine,
+} from './ExpandedRoute.styles';
+
+export function ExpandedRoute(props: PropTypes) {
```
Comment (RanGojo) [inline widget/ui/src/components/ExpandedRoute/ExpandedRoute.tsx:29]: Please double check if you handle step warning/error case.

Screenshot 2024-02-20 at 10 37 49

Screenshot 2024-02-20 at 10 37 31

**Review by RanGojo — CHANGES_REQUESTED** [review]

**Comment by nikaaru** [comment]
image

small size seems to be correct instead of xmedium

**Comment by nikaaru** [comment]
image

It seems that some text sizes are not correct.

**Comment by nikaaru** [comment]
image
The color in the parentheses does not match the figma

**Comment by nikaaru** [comment]
image
The color of the error icon does not match the figma

File: widget/embedded/src/components/Quote/Quote.tsx [hunk 4]
```diff
@@ -56,13 +65,14 @@ import {
...
   summaryHeaderStyles,
   summaryStyles,
-  Tag,
   TagContainer,
 } from './Quote.styles';
 import { QuoteCostDetails } from './QuoteCostDetails';
 import { QuoteSummary } from './QuoteSummary';
 import { QuoteTrigger } from './QuoteTrigger ';
 
+const GAS_FEE_MAX = 30;
```
Comment (nikaaru) [inline widget/embedded/src/components/Quote/Quote.tsx:74]: NIT:  I think it is better to be in this file: src/constants/quote.ts

File: widget/embedded/src/pages/Routes.tsx [hunk 5]
```diff
@@ -65,6 +65,14 @@ export function RoutesPage() {
   };
   const settings_url = `../${navigationRoutes.settings}`;
   const wallets_url = `../${navigationRoutes.wallets}`;
+
+  // TODO: It will be removed before merge
+  useEffect(() => {
```
Comment (nikaaru) [inline widget/embedded/src/pages/Routes.tsx:70]: To remind

File: widget/ui/src/components/ExpandedRoute/ExpandedRoute.tsx [hunk 6]
```diff
@@ -0,0 +1,151 @@
+import type { PropTypes } from './ExpandedRoute.types';
+
+import React from 'react';
+import { ErrorIcon, WarningIcon } from 'src/icons';
```
Comment (nikaaru) [inline widget/ui/src/components/ExpandedRoute/ExpandedRoute.tsx:4]: When run storybook (yarn dev:ui), 
we have this error:
Error: Cannot find module 'src/icons'

**Review by nikaaru — APPROVED** [review]

**Pull request merged**